app/modules/vote.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import { api } from 'neon-js'
+
+import {
+  showErrorNotification,
+  showInfoNotification,
+  showSuccessNotification
+} from './notifications'
+import {
+  getNetwork,
+  getAddress
+} from '../core/deprecated'
+
+export const setupVote = (candidateKeys: CandidateKeys = []) => async (
+  dispatch: DispatchType,
+  getState: GetStateType
+): Promise<*> => {
+  if (!Array.isArray(candidateKeys)) {
+    throw new Error('candidateKeys must be an array.')
+  }
+
+  const state = getState()
+  const address = getAddress(state)
+  const net = getNetwork(state)
+
+  dispatch(
+    showInfoNotification({ message: 'Casting vote...', autoDismiss: 0 })
+  )
+
+  try {
+    const response = await api.setupVote({
+      net,
+      account: address,
+      candidateKeys
+    })
+    if (response) {
```

Comment:
what if there was no error to catch but the response is a failure

Reply:
We should probably handle that edge case here